templates/default/oauth2_proxy.conf.erb

```diff
@@ -30,6 +30,10 @@ pass_host_header = <%= @pass_host_header.to_s %>
 ## Pass OAuth Access token to upstream via "X-Forwarded-Access-Token"
 pass_access_token = <%= @pass_access_token.to_s %>
 
+# TLS
```

Comment:
Should we make this conditional on whether `tls_cert_file`/`tls_key_file` are defined? If you're not going to bother, then I would make the `tls` resource props optional

Reply:
I made optional based on `tls_cert_file` but seems like we if we enable only https target group, it will only look for tls mode.